src/js/map/point_data/cluster.js

```diff
@@ -0,0 +1,156 @@
+import {Component, setPopupStyle, ThemeStyle} from '../../utils';
+
+const POPUP_OFFSET = [20, 0];
+
+export class Cluster extends Component {
+    constructor(parent, map, config) {
+        super(parent);
+
+        this.map = map;
+        this.tooltipItem = $('.facility-tooltip.is--cluster')[0].cloneNode(true);
+        this.config = config;
+
+        this.initClustering();
+    }
+
+    initClustering() {
+        this.markers = L.markerClusterGroup({
+            iconCreateFunction: (cluster) => this.createClusterIcon(cluster),
+            chunkedLoading: true
+        }).on('clustermouseover', (e) => {
+            this.showPopup(e.layer);
+        });
+    }
+
+    isClusteringEnabled() {
+        let enabled = false;
+        let pmConfig = this.config.config.point_markers;
+
+        if (pmConfig !== undefined && pmConfig.clustering !== undefined && pmConfig.clustering.enabled !== undefined) {
+            enabled = pmConfig.clustering.enabled;
+        }
+
+        return enabled;
+    }
+
+    createClusterIcon(cluster) {
+        const markerCount = cluster.getChildCount();
+        let colors = this.groupChildrenMarkers(cluster, true);
+
+        let circles = '';
+        let total = 0;
+        colors.forEach((c, i) => {
+            const calc = (markerCount - total) / markerCount * 100;
+            circles += `<circle 
+                            stroke-dasharray="${calc} 100" 
+                            r="16" 
+                            cx="16" 
+                            cy="16" 
+                            stroke="${c.color}" 
+                            fill="none" 
+                            stroke-width="35"></circle>`;
+            total += c.count;
+        })
+
+        const html = `<svg 
+                        width="35px" 
+                        style="border-radius:50%; background:#3f51b5;"
+                        viewBox="0 0 35 35">
+                            ${circles}
+                            <text 
+                                x="50%" 
+                                y="50%" 
+                                alignment-baseline="middle"
+                                text-anchor="middle" 
+                                dominant-baseline="middle"
+                                fill="#fff"
+                                font-size="10px" 
+                                font-weight="bold"
+                            >${markerCount}</text>
+                        </svg>`;
```

Comment:
I think it would be better to clone this HTML

Reply:
this html does not exist in the webflow export. and it is not something that the designers can manage. that's why I think we don't need to put this into index.html to be cloned.